crates/core/tedge_write/src/api.rs

```diff
@@ -33,24 +33,30 @@ impl<'a> CopyOptions<'a> {
     ///
     /// Stdin and Stdout are UTF-8.
     pub fn copy(self) -> anyhow::Result<()> {
-        let output = self
-            .command()?
-            .output()
-            .context("Starting tedge-write process failed")?;
+        let mut command = self.command()?;
+
+        let output = command.output();
+
+        let program = command.get_program().to_string_lossy();
+        let output = output.with_context(|| format!("failed to start process '{program}'"))?;
 
         if !output.status.success() {
+            let stderr = String::from_utf8(output.stderr).expect("output should be utf-8");
```

Comment:
I'm assuming that we're using `expect` here as we know that the output of `tedge-write` process can't be non-UTF8.